addons/mail/static/src/components/message/message.xml

```diff
@@ -111,54 +112,111 @@
-                                    </t>
+                                        </t>
+                                        <t t-else="">
+                                            <Popover>
+                                                <span class="o_Message_notificationIconClickable text-600 o_cursor_pointer">
+                                                    <i name="notificationIcon" class="o_Message_notificationIcon fa fa-envelope-o"/>
+                                                </span>
+                                                <t t-set-slot="opened">
+                                                    <MessageNotificationPopoverContent record="messageView"/>
+                                                </t>
+                                            </Popover>
+                                        </t>
+                                    </div>
                                 </t>
                             </div>
                         </t>
-                        <t t-if="messageView.message.subject and !messageView.message.isSubjectSimilarToOriginThreadName">
-                            <em class="o_Message_subject mx-2 mb-1">Subject: <t t-esc="messageView.message.subject"/></em>
-                        </t>
-                        <div class="o_Message_content mx-2 text-break" t-ref="content">
-                            <t t-if="!messageView.composerViewInEditing">
-                                <div class="o_Message_prettyBody" t-ref="prettyBody"/><!-- messageView.message.prettyBody is inserted here from _update() -->
-                            </t>
-                            <t t-if="messageView.composerViewInEditing">
-                                <Composer
-                                    className="'o_Message_composer mb-1'"
-                                    record="messageView.composerViewInEditing"
+                        <div 
+                            class="o_Message_bubble_wrap position-relative d-flex align-items-start"
+                            t-att-class="{
+                                'd-flex justify-content-end': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor,
+                                'pe-5': messageView.inChatWindow and !messageView.message.isCurrentUserOrGuestAuthor and !messageView.composerViewInEditing,
+                                'ps-5': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor and !messageView.composerViewInEditing,
+                                'pe-2': messageView.inChatWindow and !messageView.message.isCurrentUserOrGuestAuthor and messageView.composerViewInEditing,
+                                'ps-2': messageView.inChatWindow and messageView.message.isCurrentUserOrGuestAuthor and messageView.composerViewInEditing,
+                                'o_Message_can_be_deleted': messageView.message.canBeDeleted,
```

Comment:
The template would be easier to understand conditional styling if you introduce concept of left/right alignment in models, e.g. `MessageView/isAlignedLeft` or `MessageView/isAlignedRight`.

**This comment applies to whole commit**

Reply:
Ok for me, field name is a bit long, but I much prefer very explicit name than ambiguous or missing information, so that's good enough!